daemon/cmd/daemon_main.go

```diff
@@ -1370,6 +1373,12 @@ func runDaemon() {
 	metricsErrs := initMetrics()
 
 	d.startAgentHealthHTTPService(fmt.Sprintf("localhost:%d", option.Config.AgentHealthPort))
+	if option.Config.KubeProxyHealthzBindAddr != "" {
+		if option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict ||
+			option.Config.KubeProxyReplacement == option.KubeProxyReplacementProbe {
+			d.startAgentHealthHTTPService(fmt.Sprintf("%s", option.Config.KubeProxyHealthzBindAddr))
```

Comment:
Did you check whether a return from the http health service is the same as Kubernetes kube-proxy /healthz?

Reply:
I don't think I have checked that?

Reply:
How do I check that?

Reply:
Just run kube-proxy without Cilium, and then with curl send a request to its healthz endpoint.

Reply:
@brb This is what I am seeing.
```
#curl [::]:10256/healthz
{"lastUpdated": "2020-06-03 23:59:53.688063083 +0000 UTC m=+202.631797660","currentTime": "2020-06-03 23:59:53.688063083 +0000 UTC m=+202.631797660"}
#curl localhost:10256/healthz
{"lastUpdated": "2020-06-04 00:00:04.688465078 +0000 UTC m=+213.632199641","currentTime": "2020-06-04 00:00:04.688465078 +0000 UTC m=+213.632199641"}

Reply:
And what is cilium-agent /healthz reply?

Reply:
When do the query with cilium-agent /healthz
#curl localhost:10256/healthz
It just returns without any error.

Reply:
I think the point of this question is that for this healthz check to provide a drop-in replacement for the behaviour of kube-proxy, the API should respond in the same way to the same requests.

(I'm not familiar with the semantics / expectations of the `lastUpdated` / `currentTime` fields returned here but I'd imagine we need to come up with some kind of values for these, presumably with the equivalent behaviour for this implementation)

Reply:
@joestringer Yes I agree. Otherwise it does not solve the purpose.

Reply:
@joestringer in the case of kube-proxy service in kubenetes, the lastUpdated/currentTime is getting generated based on what time the service was queued and what time the service was updated. In our case in the 'agenthealth.go' we just return as 'OK' for the status when /healthz is queried. So we might have to rewrite this handler and come up with an strategy to report the status similar to Kubernetes. 
```
https://github.com/kubernetes/kubernetes/blob/323f34858de18b862d43c40b2cced65ad8e24052/pkg/proxy/healthcheck/proxier_health.go#L119

Reply:
@brb @joestringer  In the cilium-agent we check for the status of the Cilium alone and then quit. But also in the kubernetes proxier_health.go they check for the health in a different way and if the health is ok, they then return the lastUpdated time and currentTime both being the same if the health status is ok, otherwise they return the actual values of the currentTime and lastUpdated.
So I followed the similar approach to add log the time along with the "OK" return for status.
```
cilium$ curl 0.0.0.0:10256/healthz
{"lastUpdated": "2020-06-10 01:19:19.818577874 +0000 UTC m=+31.568753514", "currentTime": "2020-06-10 01:19:19.818577874 +0000 UTC m=+31.568753514"}
cilium$ curl [::]:10256/healthz
{"lastUpdated": "2020-06-10 01:19:55.154893031 +0000 UTC m=+66.905068676", "currentTime": "2020-06-10 01:19:55.154893031 +0000 UTC m=+66.905068676"}
cilium$ curl localhost:10256/healthz
{"lastUpdated": "2020-06-10 01:20:08.244914753 +0000 UTC m=+79.995090402", "currentTime": "2020-06-10 01:20:08.244914753 +0000 UTC m=+79.995090402"}
```
Let me know your thoughts.

Reply:
For kube-proxy health check I think we should follow the kube-proxy's logic. I.e. `lastUpdated` should store a datetime of the latest k8s service update.

Reply:
@brb Where can we get that the K8s service update info from the cilium-agent?

Reply:
We could add it to `pkg/services/services.go`, and update it when `UpsertService()` is called.